### PR TITLE
chore: Initialize jsondb with more secondary indexes

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java [hunk 1]
```diff
@@ -63,4 +86,16 @@ public SqlJsonDB realTimeDB(DBI dbi) {
         return jsondb;
     }
 
+    private void addIndex(ArrayList<Index> indexes, Kind kind, IndexedProperty indexedProperty) {
```
Comment (zregvart) [inline app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java]: ArrayList -> List

File: app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java [hunk 2]
```diff
@@ -63,4 +86,16 @@ public SqlJsonDB realTimeDB(DBI dbi) {
...
+        if (indexedProperty != null) {
+            indexes.add(new Index("/" + kind.getModelName() + "s", indexedProperty.value()));
+        }
+    }
+
+    private void addIndex(ArrayList<Index> indexes, Kind kind, UniqueProperty p) {
```
Comment (zregvart) [inline app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java]: ArrayList -> List

File: app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java [hunk 3]
```diff
@@ -28,29 +28,52 @@
...
+
+    /**
+     * This basically ensures that we get at least one Index created in the system so
+     * that the index list injection in the jsonDB does not fail due to no Indexes
+     * being defined.
+     *
+     * @return
+     */
+    @Bean
+    public Index fakeIndex() {
```
Comment (zregvart) [inline app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java]: Does `Optional<List<Index>>` help with injection?

File: app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java [hunk 4]
```diff
@@ -28,29 +28,52 @@
...
+        return FAKE_INDEX;
+    }
 
     @Bean
     @Autowired
     @SuppressWarnings("PMD.EmptyCatchBlock")
-    public SqlJsonDB realTimeDB(DBI dbi) {
+    public SqlJsonDB jsonDB(DBI dbi, List<Index> beanIndexes) {
```
Comment (zregvart) [inline app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java]: I think we could use `Optional<List<Index>>` or `@Autowired(required = false)` here

File: app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java [hunk 5]
```diff
@@ -28,29 +28,52 @@
...
 import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Creates and configures the main datastore
  */
 @Configuration
 public class DataStoreConfiguration {
     private static final Logger LOG = LoggerFactory.getLogger(SchemaCheck.class);
+    private static final Index FAKE_INDEX = new Index(null, null);
```
Comment (zregvart) [inline app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java]: Would be nicer without this

File: app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java (hunk 3 above)
Comment (chirino) [reply app/rest/runtime/src/main/java/io/syndesis/runtime/DataStoreConfiguration.java]: It did.

**Comment by chirino** [comment]
retest this please

**Pull request merged**